lib/class-wp-theme-json-resolver.php

```diff
@@ -0,0 +1,383 @@
+<?php
+/**
+ * Process the different data sources for site-level
+ * config and offers and API to work with them.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Class that abstracts the processing
+ * of the different data sources.
+ */
+class WP_Theme_JSON_Resolver {
```

Comment:
Would it make sense to add some tests to this class?

Reply:
I thought about that but discarded it because I'm not sure if there's a lot of value for this one. This is my rationale: if you look at the methods, all are straightforward calls from WordPress that mostly pull data from sources (files the active theme and core, user data from the CPT, etc). The important logic is in `WP_Theme_JSON`, for which I already wrote extensive testing.

However, I'm always happy to reconsider it, adding tests is always fun. What would you think needs testing and is not covered yet?